### PR TITLE
Make Yi.Tag use Yi.CompletionTree

### PR DESCRIPTION
**Pull request by noughtmare**
I will add some tests for this tomorrow.

### CODE REVIEW

File: yi-core/src/Yi/Tag.hs [hunk 1]
```diff
@@ -120,21 +120,20 @@ importTagTable filename = do
...
-                    -- TODO either change word-trie to use Text or
-                    -- figure out a better way all together for this
-                  , tagTrie = Trie.fromList . map (T.unpack . _unTag) $ keys cts
+                  , tagCompletionTree = CT.fromList . map (_unTag) $ keys cts
                   }
 
 -- | Gives all the possible expanded tags that could match a given @prefix@
 hintTags :: TagTable -> T.Text -> [T.Text]
-hintTags tags prefix = map (T.append prefix . T.pack) sufs
+hintTags tags prefix = map (prefix `T.append`) sufs
```
Comment (jhance) [inline yi-core/src/Yi/Tag.hs]: Can we make a function in `CompletionTree` like `full -> CompletionTree full -> [full]` that acts like this does?

Comment (noughtmare) [inline yi-core/src/Yi/Tag.hs]: I think there should be "higher" level functions like this in other modules like `Yi.TextCompletion` and `Yi.Completion`.

Comment (jhance) [inline yi-core/src/Yi/Tag.hs]: Alright, fine to defer this then

**Comment by jhance** [comment]
Verified that `:tag` still works, I feel confident enough to push this now. (Tests are still nice though)

**Pull request closed without merging**